### PR TITLE
Add ability to release dev tag

### PR DESCRIPTION
**Pull request by krzysztofzuraw**
I want to merge this change because it adds ability to release macaw-ui as dev tag if you add label on PR.

This PR closes #...

### Pull Request Checklist

- [ ] New components are exported from `./src/components/index.ts`.
- [ ] The storybook story is created and documentation is properly generated.
- [ ] New component is wrapped in `forwardRef`.

### CODE REVIEW

**Review by stmpn — APPROVED** [review]

File: .github/workflows/dev-release.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    if: ${{ github.event.label.name == 'release dev tag' }}
+    runs-on: ubuntu-22.04
+    env:
+      HUSKY: 0
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Setup PNPM
+        uses: pnpm/action-setup@a3252b78c470c02df07e9d59298aecedc3ccdd6d # v3.0.0
```
Comment (witoszekdev) [inline .github/workflows/dev-release.yml:17]: question: why do we need this specific hash? Can't we use `version` option to use specific `pnpm` version and use latest GH action?

CleanShot 2024-03-15 at 11 41 54@2x

or it doesn't work for 3.x.x`? 😅

**Review by poulch — APPROVED** [review]

File: .github/workflows/dev-release.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        with:
+          run_install: |
+            - args: [--frozen-lockfile]
+      - name: Check for changeset
+        run: pnpm exec changeset status --since origin/main
+      - name: Create .npmrc
+        run: |
+          cat << EOF > "$HOME/.npmrc"
+            //registry.npmjs.org/:_authToken=$NPM_TOKEN
+          EOF
```
Comment (witoszekdev) [inline .github/workflows/dev-release.yml:27]: question: Can this action be run for external contributors? I can see that someone adds additional action that reads this file and steals our NPM_TOKEN 🙈

**Review by witoszekdev — APPROVED** [review]

File: .github/workflows/dev-release.yml (hunk 2 above)
Comment (krzysztofzuraw) [reply .github/workflows/dev-release.yml:27]: I changed this setting - so we have to grant access for people outside of Saleor so then can run our workflows on their forks:

CleanShot 2024-03-15 at 12 06 58@2x

File: .github/workflows/dev-release.yml (hunk 1 above)
Comment (krzysztofzuraw) [reply .github/workflows/dev-release.yml:17]: This is pinning github action for PNPM not version of PNPM inself 😄 . We added it based on GitHub recommendation https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

Comment (witoszekdev) [reply .github/workflows/dev-release.yml:17]: Nice, I didn't know about this once ❤️

**Pull request merged**